packages/eslint-plugin/src/rules/sort-ngmodule-arrays.ts

```diff
@@ -0,0 +1,75 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+import { MODULE_CLASS_DECORATOR } from '../utils/selectors';
+import {
+  getDecoratorPropertyValue,
+  isArrayExpression,
+  isIdentifier,
+} from '../utils/utils';
+
+import { NgModule } from '@angular/compiler/src/core';
+
+type Options = [];
+export const RULE_NAME = 'sort-ngmodule-arrays';
+export type MessageIds = 'sortedFailure';
+const validProperties: (keyof NgModule)[] = [
+  'bootstrap',
+  'declarations',
+  'entryComponents',
+  'exports',
+  'imports',
+  'providers',
+  'schemas',
+];
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'NgModule property arrays should be alphabetized for easy visual scanning.',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      sortedFailure:
+        'NgModule property arrays should be sorted in ASC alphabetical order.',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      [MODULE_CLASS_DECORATOR](node: TSESTree.Decorator) {
+        validProperties.forEach((prop: keyof NgModule) => {
+          const initializer = getDecoratorPropertyValue(node, prop);
+          if (
+            !initializer ||
+            !isArrayExpression(initializer) ||
+            initializer.elements.length < 2
+          ) {
+            return;
+          }
+          const elements = initializer.elements.filter(isIdentifier);
+
+          elements.some((e, i, list) => {
+            if (i === 0) {
+              return false;
+            }
+            const previous = list[i - 1];
+            if (e.name.localeCompare(previous.name) === -1) {
+              context.report({
+                node: previous,
+                messageId: 'sortedFailure',
+              });
+              return true;
+            } else {
+              return false;
+            }
+          });
```

Comment:
What about using `find` instead of `some` and then use this element (if defined) later? Here's a suggestion:

```suggestion
          const unorderedNode = initializer.elements
            .filter(isIdentifier)
            .find(
              ({ name }, index, list) =>
                name.localeCompare(list[index + 1]?.name) === 1,
            );
          if (!unorderedNode) return;
          context.report({ messageId, node: unorderedNode });
```